pkg/ebpf/c/capture_filtering.h

```diff
@@ -93,15 +99,41 @@ statfunc bool filter_file_type(void *ctx, void *filter_map, size_t map_idx, stru
             if (pipe != NULL) {
                 type_filter_match = true;
             }
-        } else if (*type_filter & FILTER_SOCKET_FILES) {
+        }
+        if (*type_filter & FILTER_SOCKET_FILES) {
             if (imode_mode & S_IFSOCK) {
                 type_filter_match = true;
             }
-        } else if (*type_filter & FILTER_NORMAL_FILES) {
+        }
+        if (*type_filter & FILTER_NORMAL_FILES) {
             if (imode_mode & S_IFREG) {
                 type_filter_match = true;
             }
         }
+        if (*type_filter & FILTER_ELF_FILES) {
+            file_id_t file_id = get_file_id(file);
+            file_id.ctime = 0;
+            if (start_pos == 0) {
+                // Check if header is matching ELF header
+                u8 header[FILE_MAGIC_HDR_SIZE] = {};
+                fill_file_header(io_data, header);
```

Comment:
nit: usually destination (header) argument comes before source (io_data) argument, so I would switch the arguments order to avoid confusion